aspnetcore/native-aot/index.md

```diff
@@ -0,0 +1,190 @@
+    <ServerGarbageCollection>false</ServerGarbageCollection>
+    <InvariantGlobalization>true</InvariantGlobalization>
+    <PublishAot>true</PublishAot>
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles> <!-- Added to emit generated files -->
+  </PropertyGroup>
+
+</Project>
+```
+
+With the project file updated, run the ```dotnet build``` command (```publish``` isn't necessary to view generated code). In the built output file there will be a ```obj/Debug/net8.0/generated/``` directory which contains all the generated files for the project.
+
+When ```dotnet publish``` is run, the project source files and generated source files are compiled as normal and then outputted assemblies are passed into an native IL linker which produces the native executable which no longer requires the .NET runtime to be installed.
+
+## Benefits of using .NET native AOT with ASP.NET Core
+
+Using the .NET native AOT deployment model provides the following benefits:
+
+* **Minimize disk footprint**; when publishing using native AOT a single executable is produced containing just the code from external dependencies that is used to support the program. Reduced executable size can lead to smaller container images (in containerized deployment scenarios) which can reduce deployment time.
+* **Reduced startup time**; native AOT applications can show reduced start-up times which means the application is ready to service requests quicker. This can also help during deployment where container orchestrators need manage transition from one version of the application to another.
+* **Reduce memory demand**; native AOT applications can have reduced memory demands depending on the nature of the work being performed by the application. This reduced memory consumption can lead to greater deployment density and improved scalability.
+
+## When using .NET native AOT with ASP.NET Core should be avoided
+
+Not all features in ASP.NET Core are currently compatible with .NET native AOT. The following table summarizes ASP.NET Core feature compatibility with .NET native AOT:
+
+| Feature                       | Fully Supported | Partially Supported | Not Supported        |
+| ----------------------------- | --------------- | ------------------- | -------------------- |
+| Minimal APIs                  |                 | Yes                 |                      |
+| MVC                           |                 |                     | No                   |
+| Blazor                        |                 |                     | No                   |
+| SignalR                       |                 |                     | No                   |
+| Authentication                |                 |                     | No (JWT coming soon) |
+
+It is important to test application functionality thoroughly when moving to a native AOT deployment model to ensure that functionality observed during developmnet (using the runtime) is preserved in the native executable.
```

Comment:
We should also talk about AOT warnings here. Basically if you get any warnings, your app is not guaranteed to work correctly. If you get don't get any AOT warnings at publish-time, you should be confident that your application will work the same after publishing for AOT as it did during F5 / `dotnet run` time.